File: Searching/Linear_search/linear_search.cpp
#include <iostream>
#include <vector>


void linear_search(std::vector<int>& arr, int& key)
{

	bool found = false;
	int index_of_element = -1;

	for (auto& v: arr)
	{
		index_of_element++;

		if (v == key)
		{

			found = true;
			break;

		}
	}

	std::cout << "Element " << key <<" ";

	if (found)
		std::cout << "found at index: " << index_of_element << std::endl;
	else
		std::cout << "not found." << std::endl;


}


int main()
{

	std::vector<int> darray;

	int vertices_length;

	int element;

	int key;
	std::cout << "Enter the no of elements: " << std::endl;

	std::cin >> vertices_length;

	std::cout << "Enter the element(s): " << std::endl;
	while (vertices_length--)
	{

		std::cin >> element;
		darray.push_back(element);

	}
	std::cout << "Enter the element to be searched: " << std::endl;
	std::cin >> key;

	linear_search(darray, key);
	return 0;
}

File: Searching/Linear_search/linear_search.py
arr = []


def accept_user_input():
    
    global arr
    
    print("Enter the elements: ", end="\n")
    arr = list(input().split())
    arr = [int(i) for i in arr]
    

def linear_search(arr, key):
    
    found = False
    index_of_element = -1
    
    for i in arr:
        
        index_of_element += 1
        
        if i == key:
            
            found = True
            break
            
    print("Element ", key, end="")
            
    if (found):
        
        print(" found at index: ", index_of_element)
    
    else:
         
        print(" not found.") 
                      

accept_user_input()

key = int(input("Enter the search element: "))

linear_search(arr, key)    
